libs/store/src/lib/infrastructure/persistence/store/entity-store.ts

```diff
@@ -90,7 +99,7 @@ export class DefaultQueryService<TModel> extends QueryService<TModel> {
 
     getWithQuery(query: QuerySnapshot<TModel>): Observable<TModel[]> {
         // Pass query payload instead of QueryParams
-        // TODO: Should we override EntityCollectionService interface for that? 
+        // TODO: Should we override EntityCollectionService interface for that?
         return this.service.getWithQuery(query as any);
```

Comment:
We need to map the return data here (and within the `getByKey` method) from the EntityCollectionService to call `instanceForType`, so that the end-user interacts with Entity objects and not DTOs.